src/main/java/io/confluent/copycat/hdfs/HdfsWriter.java

```diff
@@ -224,6 +260,8 @@ private void execute(TopicPartition topicPart) throws IOException {
       } catch (IllegalWorkerStateException e) {
         // Should we retry in this case?
         throw e;
+      } catch (SchemaProjectorException e) {
+        throw e;
```

Comment:
Why this separate block for `SchemaProjectorException`? Isn't that just a CopycatException that could be handled by the existing block? Or are you throwing the exception because this is effectively unrecoverable?